google-cloud-spanner/lib/google/cloud/spanner/service.rb

```diff
@@ -330,19 +333,42 @@ def streaming_read_table session_path, table_name, columns, id: nil,
           end
           execute do
             service.streaming_read \
-              session_path, table_name, columns, key_set, limit: limit
+              session_path, table_name, columns, key_set,
+              transaction: transaction, limit: limit
           end
         end
 
-        def commit session_path, mutations = []
-          execute do
+        def commit session_name, mutations = [], transaction_id: nil
+          tx_opts = nil
+          if transaction_id.nil?
             tx_opts = Google::Spanner::V1::TransactionOptions.new(read_write:
               Google::Spanner::V1::TransactionOptions::ReadWrite.new)
+          end
+          execute do
             service.commit \
-              session_path, mutations, single_use_transaction: tx_opts
+              session_name, mutations,
+              transaction_id: transaction_id, single_use_transaction: tx_opts
           end
         end
 
+        def begin_transaction session_name
+          tx_opts = Google::Spanner::V1::TransactionOptions.new(read_write:
+            Google::Spanner::V1::TransactionOptions::ReadWrite.new)
+          execute { service.begin_transaction session_name, tx_opts }
+        end
+
+        def create_snapshot session_name, strong: nil, timestamp: nil,
+                            staleness: nil
+          tx_opts = Google::Spanner::V1::TransactionOptions.new(read_only:
+            Google::Spanner::V1::TransactionOptions::ReadOnly.new({
+              strong: strong,
+              read_timestamp: Convert.time_to_timestamp(timestamp),
+              exact_staleness: Convert.number_to_duration(staleness),
+              return_read_timestamp: true
```

Comment:
No, those options are for single-use transactions, and as such are available on `Client#read` and not used on `Snapshot#read`.